com.vrcfury.vrcfury/Editor/VF/Service/OnDemandSyncService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using VF.Builder;
+using VF.Feature.Base;
+using VF.Injector;
+using VF.Utils.Controller;
+using VRC.SDK3.Avatars.Components;
+
+namespace VF.Service {
+    [VFService]
+    public class OnDemandSyncService {
+        [VFAutowired] private readonly AvatarManager manager;
+        [VFAutowired] private readonly MathService math;
+        [VFAutowired] private readonly DirectBlendTreeService directTree;
+        [VFAutowired] private readonly SmoothingService smoothingService;
+        
+        private readonly List<VFAFloat> onDemandSyncParameters = new List<VFAFloat>();
+        
+        public void SetParameterOnDemandSync(VFAFloat parameter) {
+            onDemandSyncParameters.Add(parameter);
+        }
+
+        [FeatureBuilderAction(FeatureOrder.OnDemandSync)]
+        public void Apply() {
+            if (onDemandSyncParameters.Count == 0) return;
+            
+            var fx = manager.GetController(VRCAvatarDescriptor.AnimLayerType.FX);
+            var layers = fx.GetRaw().GetLayers();
+            foreach (var param in onDemandSyncParameters) {
+                manager.GetParams().GetParam(param.Name()).networkSynced = false;
+            }
+            var pointer = fx.NewInt("SyncPointer", synced: true);
+            var localPointer = fx.NewInt("LocalPointer"); // Use a local only pointer to avoid race condition
+            var data = fx.NewFloat("SyncData", synced: true);
+            var sending = fx.NewFloat("IsSending");
+
+            var txLayer = fx.NewLayer("Send");
```

Comment:
It's usually a good idea to send remote users into a "remote trap" state within "local-only" layers, so they don't have to process any conditionals or anything and just remain stuck in that no-op "remote" state forever.

Reply:
The receive and send layer was merged into one and only process the needed conditionals for remote or local